### PR TITLE
fix: allow attendee multiple day checkin

### PR DESCRIPTION
**Pull request by maze-runnar**
Fixes https://github.com/fossasia/open-event-frontend/issues/2016
recreation of https://github.com/fossasia/open-event-frontend/pull/7480

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/ui-table/cell/events/view/tickets/attendees/cell-action.js [hunk 1]
```diff
@@ -1,5 +1,54 @@
...
+        const checkinDates = [];
+        for (const i of x) {
+          checkinDates.push(moment(i).format('MM-DD-YYYY'));
+        }
+        return checkinDates;
+      }
+      return [];
+    }
+
+    get checkoutTimes() {
```
Comment (MeghalBisht) [inline app/components/ui-table/cell/events/view/tickets/attendees/cell-action.js:32]: @maze-runnar  shouldn't these all be computed properties?

File: app/controllers/events/view/tickets/attendees/list.js [hunk 2]
```diff
@@ -70,16 +70,23 @@ export default class extends Controller.extend(EmberTableControllerMixin) {
...
-  toggleCheckIn(attendee_id) {
+  toggleCheckIn(attendee_id, date, isCheckedInCurrently) {
     const attendee = this.store.peekRecord('attendee', attendee_id, { backgroundReload: false });
-    attendee.toggleProperty('isCheckedIn');
-    if (attendee.isCheckedIn) {
-      const newCheckinTimes = attendee.get('checkinTimes') === null ? `${moment().toISOString()}` : `${attendee.get('checkinTimes')},${moment().toISOString()}`;
+    let myTime = moment().toISOString();
+    if (moment(date, 'MM-DD-YYYY').format('MM-DD-YYYY') !== moment().format('MM-DD-YYYY')) {
+      myTime = moment(date, 'MM-DD-YYYY');
+      myTime = myTime.format('YYYY-MM-DD') +  'T13:00:00.000Z';
```
Comment (MeghalBisht) [inline app/controllers/events/view/tickets/attendees/list.js:78]:
```suggestion
      myTime = myTime.format('YYYY-MM-DD') + 'T13:00:00.000Z';
```

File: app/components/ui-table/cell/events/view/tickets/attendees/cell-action.js (hunk 1 above)
Comment (maze-runnar) [reply app/components/ui-table/cell/events/view/tickets/attendees/cell-action.js:32]: No

**Review by MeghalBisht — APPROVED** [review]

**Comment by mariobehling** [comment]
The attendee checkin dropdown is unfortunately still not working, in particular when there is only one event day there is no menu choice.

**Pull request merged**